pennylane/_device.py

```diff
@@ -878,6 +878,15 @@ def check_validity(self, queue, observables):
 
             operation_name = o.name
 
+            if getattr(
+                o, "return_type", None
+            ) == qml.operation.MidMeasure and not self.capabilities().get(
+                "supports_mid_measure", False
+            ):
+                raise DeviceError(
+                    f"Mid-circuit measurements are not natively supported on device {self.short_name}"
```

Comment:
```suggestion
                    f"Mid-circuit measurements are not natively supported on device {self.short_name}. "
                    "Apply the @qml.defer_measurements decorator to your quantum function to "
                    "simulate the application of mid-circuit measurements on this device."
```
Just a hint on what to do